modin/core/dataframe/pandas/partitioning/partition.py

```diff
@@ -38,6 +38,24 @@ class PandasDataframePartition(ABC):  # pragma: no cover
     _width_cache = None
     _identity_cache = None
     _data = None
+    execution_wrapper = None
+
+    # these variables are intentionally initialized at runtime
+    # so as not to initialize the engine during import
+    _iloc_func = None
+
+    def __init__(self):
+        if type(self)._iloc_func is None:
+            # Places `_iloc` function into the storage to speed up
+            # remote function calls and caches the result.
+            # It also postpones engine initialization, which happens
+            # implicitly when `execution_wrapper.put` is called.
+            if self.execution_wrapper is not None:
+                type(self)._iloc_func = staticmethod(
+                    self.execution_wrapper.put(self._iloc)
+                )
```

Comment:
```suggestion
                type(self)._iloc_func = self.execution_wrapper.put(self._iloc)
```
Why do we need to pass an obj ref to staticmethod?

Reply:
Class-level `__setattr__` behaves differently and does quite a few magic tricks, one of which is when you assign an attribute which is a function, it becomes an unbound method _during assignment_, etc. Hence if no `staticmethod()` call is done here, such transformation is done, and function starts to receive an extra `self` argument when called, which leads to errors ([ref](https://github.com/modin-project/modin/actions/runs/5519619408/jobs/10065219105)).